poc_mobile_verifier/src/speedtests.rs

```diff
@@ -70,6 +71,92 @@ impl SpeedtestRollingAverage {
         }
     }
 
+    pub async fn validate_speedtests<'a>(
+        exec: impl SpeedtestStore + Copy + 'a,
+        file_store: &FileStore,
+        epoch: &'a Range<DateTime<Utc>>,
+    ) -> Result<impl Stream<Item = Self> + 'a> {
+        let speedtests = Arc::new(Mutex::new(HashMap::new()));
+        Ok(file_store
+            .source(
+                stream::iter(
+                    file_store
+                        .list_all(FileType::CellSpeedtestIngestReport, epoch.start, epoch.end)
+                        .await?,
+                )
+                .map(Ok)
+                .boxed(),
+            )
+            .filter_map(move |msg| {
+                let speedtests = speedtests.clone();
+                async move {
+                    let msg = if let Ok(msg) = msg {
```

Comment:
why not do this in the `move` section?

Reply:
The first move moves the Arc into the FnMut closure, the second moves the clone of the Arc into the async block